### PR TITLE
Lockers and crates shake when someone is trying to break out of them

### PR DESCRIPTION
**Pull request by Lufferly**
## About The Pull Request

Makes lockers and crates do a little shake when someone is trying to resist out of them.

https://github.com/tgstation/tgstation/assets/40921881/94703c79-a22c-43e6-ac31-8621173a0953

## Why It's Good For The Game

A little bit of visual flair, lets players know if someone is trying to get out of a locker.

## Changelog

:cl: Seven
add: Lockers and crates now shake when someone is attempting to resist out of them.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
I think it would be nicer if it shook a little bit and then paused in a loop, instead of vibrating constantly like a washing machine

**Comment by Lufferly** [comment]
> I think it would be nicer if it shook a little bit and then paused in a loop, instead of vibrating constantly like a washing machine

Is the new video something like what you are thinking?

**Comment by DrDiasyl** [comment]
I HAVE SEEN THIS BEING USED FOR ERP ON SOME BAY SERVERS.

On other note, very nice!

**Comment by Jacquerel** [comment]
> > I think it would be nicer if it shook a little bit and then paused in a loop, instead of vibrating constantly like a washing machine
> 
> Is the new video something like what you are thinking?

yes I like this

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 1]
```diff
@@ -1045,6 +1053,23 @@ GLOBAL_LIST_EMPTY(roundstart_station_closets)
...
+	// Assuming we decide to shake again, how long until we check to shake again
+	var/next_check_time = 1 SECONDS
+
+	// How long we shake between different calls of Shake(), so that it starts shaking and stops, instead of a steady shake
+	var/shake_duration =  0.3 SECONDS
+
+	for(var/mob/living/mob in contents)
+		if(DOING_INTERACTION_WITH_TARGET(mob, src))
+			// Shake and queue another check_if_shake
+			src.Shake(x_shake_pixel_shift, y_shake_pixel_shift, shake_duration, shake_interval = 0.1 SECONDS)
```
Comment (san7890) [inline code/game/objects/structures/crates_lockers/closets.dm:1067]:
```suggestion
			Shake(x_shake_pixel_shift, y_shake_pixel_shift, shake_duration, shake_interval = 0.1 SECONDS)
```

useless to have `src` here

**Review by san7890 — CHANGES_REQUESTED** [review]

**Pull request merged**